frappe/public/js/frappe/form/controls/datetime.js

```diff
@@ -35,19 +35,23 @@ frappe.ui.form.ControlDatetime = class ControlDatetime extends frappe.ui.form.Co
 	get_now_date() {
 		return frappe.datetime.now_datetime(true);
 	}
+	/** @param {string | null} value */
 	parse(value) {
-		if (value) {
-			value = frappe.datetime.user_to_str(value, false);
+		let parsed = typeof value === "string" ? value.trim() : "";
 
-			if (!frappe.datetime.is_system_time_zone()) {
-				value = frappe.datetime.convert_to_system_tz(value, true);
-			}
+		if (["today", "now"].includes(parsed.toLowerCase())) {
+			parsed = frappe.datetime.now_datetime(false);
```

Comment:
Wouldn’t we want to add the `as_obj` param to our helper function get_now_date()?

If we have a helper, let’s consistently use it.